classes/observers/listeners/SubmissionUpdatedListener.inc.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file classes/observers/listeners/SubmissionUpdatedListener.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionUpdatedListener
+ * @ingroup core
+ *
+ * @brief Listener fired when submission's updated
+ */
+
+namespace PKP\observers\listeners;
+
+use PKP\Jobs\Submissions\UpdateSubmissionSearchJob;
+use PKP\observers\events\BasePublicationEvent;
+
+class SubmissionUpdatedListener
+{
+    /**
+     * Handle the listener call
+     *
+     *
+     */
+    public function handle(BasePublicationEvent $event)
```

Comment:
Interesting approach, it allows to avoid code duplication; I'd probably use an event subscriber here to listen to Publish/Unpublish events separately to avoid ambiguities.

Reply:
Yes! My main idea behind here it's to avoid code duplication.

In fact, `PublishedEvent` and `UnpublishedEvent` are called in different flows, but internally they do a similar job, update the search engine. So I've opt to hook both to call `SubmissionUpdatedListener` on `PKPEventServiceProvider::$listen`, but keeping their own "personality".

Reply:
My concern here is that looking at the `SubmissionUpdatedListener::handle()` it's not clear to which event it listens. The ambiguity for me here is that it does not listen to `BasePublicationEvent` but to one of its ancestors.

Reply:
I've added a `subscribe` method on this Listener. This probably will let users know better which events are associated with it.